src/util/json_expr.cpp

```diff
@@ -19,6 +21,64 @@ Author: Peter Schrammel
 #include "fixedbv.h"
 #include "std_expr.h"
 #include "config.h"
+#include "identifier.h"
+#include "language.h"
+#include "invariant.h"
+
+#include <langapi/mode.h>
+
+static exprt simplify_json_expr(
+  const exprt &src,
+  const namespacet &ns)
+{
+  if(src.id()==ID_constant)
+  {
+    const typet &type=ns.follow(src.type());
+
+    if(type.id()==ID_pointer)
+    {
+      const irep_idt &value=to_constant_expr(src).get_value();
+
+      if(value!=ID_NULL &&
+         (value!=std::string(value.size(), '0') ||
+          !config.ansi_c.NULL_is_zero) &&
+         src.operands().size()==1 &&
+         src.op0().id()!=ID_constant)
+        // try to simplify the constant pointer
+        return simplify_json_expr(src.op0(), ns);
```

Comment:
Recommend improve this comment -- this is *discarding* the binary value  of the pointer, keeping only the symbolic expression. It is not a simplification per se.